test/suite/test_hs11.py

```diff
@@ -115,7 +159,97 @@ def test_non_ts_updates_clears_hs(self):
                         else:
                             self.assertEqual(cursor[self.create_key(i)], value2)
                     else:
-                        self.assertEqual(cursor[self.create_key(i)], value1)
+                        if ts == 5 and self.modify and self.value_format != '8t':
+                            self.assertEqual(cursor[self.create_key(i)], mod_value)
+                        else:
+                            self.assertEqual(cursor[self.create_key(i)], value1)
+
+        if self.update_type == 'deletion':
+            hs_truncate = self.get_stat(stat.conn.cache_hs_key_truncate_onpage_removal)
+            self.assertGreater(hs_truncate, 0)
+
+    def test_non_ts_updates_clears_hs_small_rows(self):
```

Comment:
@kommiharibabu I was thinking we could change this to be a scenario of the above test. Thoughts?

Reply:
Nice idea. Yes, by controlling the number of rows based on the scenario. It will simplify the test.